File: container05_php/html/checkout.php
<?php
session_start();
if (!isset($_SESSION["user"])) {
    header("Location: login.php");
    exit;
}
$db = new SQLite3('ecommerce.db');
$product_id = $_POST["product_id"] ?? null;
$product = $db->querySingle("SELECT nome FROM produtos WHERE id = " . intval($product_id));
?>
<h2>Checkout</h2>
<form method="POST" action="confirm.php">
    <p>Produto: <?= htmlspecialchars($product) ?></p>
    Endereço: <input type="text" name="address" required><br>
    <input type="hidden" name="product_id" value="<?= htmlspecialchars($product_id) ?>">
    <input type="submit" value="Finalizar Compra">
</form>

File: container02_cgi/cgi-bin/checkout.pl
#!/usr/bin/perl
use CGI;
use strict;

my $q = CGI->new;
print $q->header;
print $q->start_html("Checkout");

my $produto = $q->param("produto");

print "<h2>Checkout</h2>";
print "<form method='POST' action='/cgi-bin/confirm.pl'>";
print "<input type='hidden' name='produto' value='$produto'>";
print "Endereço: <input type='text' name='endereco'><br>";
print "<input type='submit' value='Finalizar Compra'>";
print "</form>";
print $q->end_html;
